protocol_details/system_considerations.rst

```diff
@@ -2,41 +2,80 @@
+   To develop an AMP system using the OpenAMP Framework, it is important to determine the memory
+   regions that would be owned and shared between each of the participating software environments in
+   the AMP system. For example, in a configuration such as this, the memory address ranges owned
+   (for code/data/bss/heap) by each participating OS or bare metal context, and the shared memory
+   regions to be used by IPC mechanisms (virtio rings and memory for data buffers) need to be
+   determined. Memory alignment requirements should be taken into consideration while making this
+   determination.
 
-   The following image illustrates the memory layout for Linux host/RTOS-based remote application, and RTOS-based host/bare metal-based remote application in chain configuration.
+   The following image illustrates the memory layout for Linux host/RTOS-based remote application,
+   and RTOS-based host/bare metal-based remote application in chain configuration.
 
    .. image:: ../images/memory_layout.jpg
 
    - Ensure cooperative usage of shared resources between software environments in the AMP system
 
-   For the purpose of this discussion, assume you are using a Linux host/bare metal- based remote system configuration.
-
-   The interrupt controller is typically a shared resource in multicore SoCs. It is general practice for OSs to reset and initialize (clear and disable all interrupts) the interrupt controller during their boot sequence given the general assumption that the OS would own the entire system. This will not work in AMP systems; if an OS in remote software context resets and initializes the interrupt controller, it would catastrophically break the host controller software context's runtime since the host context could already be using the interrupt controller to manage its interrupt resources. Therefore, remote software environments should be patched such that they cooperatively use the interrupt controller (for example, do not reset/clear/disable all interrupts blindly but initialize only the interrupts that belong to the remote context). Ensure the timer peripheral used by the remote OS/RTOS context is different from the one used by the host controller software context so the individual runtimes do not interfere with each other.
+   For the purpose of this discussion, assume you are using a Linux host/bare metal- based remote
+   system configuration.
+
+   The interrupt controller is typically a shared resource in multicore SoCs. It is general practice
```

Comment:
Some of the paragraphs are too long. Consider splitting at "Therefore, remote...".

Reply:
done in the commit added to the PR